explainer.md

```diff
@@ -48,8 +46,8 @@ Date: 2019-07-17
 
 The **Badging API** is a proposed Web Platform API allowing websites to apply
 badges (small status indicators) to pages or sets of pages on their origin. We
-are deliberately agnostic about which contexts a badge can appear in, but have
-two fairly different contexts in mind:
+are deliberately agnostic about which contexts a badge can appear in, but there
+are two major categories of context that will be supported:
```

Comment:
```suggestion
are two major categories of context that we would like to support:
```

As this feature is UI heavy, I need the document to be framed in a manner that I can pitch internally to our Firefox Product Team **as a proposal**. I.e., it can't be read like "this is how it's going to be", as the Product team will actually decided what gets implemented in Firefox. 

I know this is different than how Google/Chrome teams work... and Firefox may opt not to support the scoping stuff if we find it too complicated to implement/invest time in over other web features.

Reply:
Note that "supported" here refers to the functionality that would be supported if everything in this document was implemented. It doesn't make any claims about implementations supporting it. And these two sides to the API are designed to be implementable independently (neither reqiures the other). So I don't see any conflict here with Firefox's plan to not (initially) support scoping side of the API.

I'm not sure if this is just about this sentence, or more general. In general, I don't see the point of rewriting "we will do X" sentences as "we would like to do X". By definition, every sentence in the proposal is conditional on the proposal being accepted.

To resolve this, I've rewritten it as "we are specifying methods to handle two major different categories of context".

> Firefox may opt not to support the scoping stuff if we find it too complicated to implement/invest time in over other web features.

I think that would be fine, if you don't plan to support app badging (just document badging). That is the intention: if you are only badging documents while they are open, just implement the "ForDocument" methods. If you want to have persistent badges on shortcuts to websites while the site itself is not open (e.g., installed apps, but you could also apply the badge to other shortcuts such as links on the desktop, bookmarks, etc), then you implement the "ForScope" methods.

For that latter use case, we are all open to any proposal that lets us get away with not tying the badge to a URL scope. But nobody has suggested any. This is not a solution in search of a problem. We have a very clear problem: allow a site to set a semi-permanent badge on the icon of an installed web app, which persists even after the last window closes, and can be set by a service worker when no windows are open. Scoped URLs is our take on the solution.